src/platform/intel/cavs/include/cavs/lib/shim.h

```diff
@@ -49,13 +49,13 @@ static inline void shim_write64(uint32_t reg, uint64_t val)
 
 static inline uint32_t sw_reg_read(uint32_t reg)
 {
-	return *((volatile uint32_t*)((SRAM_SW_REG_BASE -
+	return *((volatile uint32_t*)(((uintptr_t)SRAM_SW_REG_BASE -
```

Comment:
Remove all `( )` parentheses:
```suggestion
const uintptr_t addr = SRAM_SW_REG_BASE - SRAM_ALIAS_OFFSET + reg;
return *(volatile uint32_t *)addr;
```